### PR TITLE
Use go code to create/delete webhook

### LINKED ISSUE #327: Use a go library/3rd party open source code to create webhook
Currently the webhook is created using a Task - this is time consuming and difficult to monitor.  Change to using a go library to create the webhook in github and take cleanup actions if a failure occurs.

For example, if the webhook has been added to the eventlistener and then creation of the webhook fails, we should also remove the entry from the event listener.

Remove the Task yaml from the config directory and all the yaml files in which it is embedded.  Test code can also be removed from shared-test-funcs that creates/returns dummy reaults for taskruns.

### PR DESCRIPTION
**Pull request by ncskier**
# Changes
Fixes #327
Use go code to create/delete webhooks instead of using a TaskRun.
To create/delete the webhook, we subscribe/unsubscribe using the GitHub
PubSubHubbub API. More info about the API here:
https://developer.github.com/v3/repos/hooks/#pubsubhubbub

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by dibbles** [comment]
yes - remove the task and also delete using the git client

File: webhooks-extension/pkg/endpoints/github.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+// getGitHubHubbubAPI returns the API URL for the GitHub PubSubHubbub API
+func getGitHubHubbubAPI(u *url.URL) string {
+	// Public GitHub PubSubHubbub API URL is "https://api.github.com/hub"
+	hubbubAPI := "https://api.github.com/hub"
+
+	// Enterprise GitHub API URL is "https://github.company.com/api/v3/hub"
+	if company := getGitHubEnterpriseCompany(u); company != "" {
+		hubbubAPI = fmt.Sprintf("https://github.%s.com/api/v3/hub", company)
```
Comment (dibbles) [inline webhooks-extension/pkg/endpoints/github.go]: Implies if you use github enterprise your server api must be github.company.com/api/v3  ...... could a company not use any dns hostname they want?  Is this a guideline given by GitHub for GHE installs?  Do we need to consider alternatives?

Comment (ncskier) [reply webhooks-extension/pkg/endpoints/github.go]: Thanks Duane, good catch! I've updated the code as well as the tests to catch these cases (https://github.com/tektoncd/experimental/pull/351/files#diff-e4756d2adeff9f4613bce3d5106b8b37R108). Please let me know if there's anything else I missed!

**Comment by ncskier** [comment]
Thanks for the review Duane! I've removed all instances of that create-webhook Task, and I've updated the git URL processing to catch the GHE case you mentioned.

Also, here's the issue I created for moving from the PubSubHubbub API to the Hooks API: https://github.com/tektoncd/experimental/issues/362.

Let me know if there are any more changes I can make.

**Pull request merged**